python/pyiceberg/io/pyarrow.py

```diff
@@ -165,6 +169,24 @@ def to_input_file(self) -> "PyArrowFile":
 
 
 class PyArrowFileIO(FileIO):
+    def __init__(self, properties: Properties = EMPTY_DICT):
+        self.get_fs_and_path: Callable = lru_cache(self._get_fs_and_path)
+        super().__init__(properties=properties)
+
+    def _get_fs_and_path(self, location: str) -> Tuple[FileSystem, str]:
+        uri = urlparse(location)  # Create a ParseResult from the URI
+        if not uri.scheme:  # If no scheme, assume the path is to a local file
+            return FileSystem.from_uri(os.path.abspath(location))
+        elif uri.scheme in {"s3", "s3a", "s3n"}:
```

Comment:
Since the pyarrow [filesystem interface](https://arrow.apache.org/docs/python/filesystems.html) support GCS and HDFS, should we add more `elif` blocks here to generate either of those filesystems?

Reply:
I suspect we'll want to do that in the future, but I don't know enough about either to what schemes we need to match on and what properties to pass through. (I've personally only ever worked with file:// and s3://)

Perhaps it makes sense to do that in a later PR?

Reply:
I agree that we should do that in a separate PR to keep this one nice and small.

Reply:
+1 I agree